File: courseify-backend/src/main/kotlin/sorsix/project/courseify/domain/Course.kt
package sorsix.project.courseify.domain

import jakarta.persistence.*

@Entity
@Table(name = "course")
data class Course(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,

    val title: String,

    val description: String,

    val thumbnail: String,

    @ManyToOne
    @JoinColumn(name = "author_id")
    val author: User,

    val isActive: Boolean = false

) {
}

File: courseify-frontend/src/app/interfaces/course.ts
import { Category } from "./Category";


export interface Course {
  id: number;
  title: string;
  description: string;
  author: Author;
  category: Category;
  thumbnail: string;
  isActive: boolean;
}

interface Author {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  username: string;
}

